### PR TITLE
Add MORE missing entity API

### PR DESCRIPTION
**Pull request by Owen1212055**
Adds:

Bee rolling override (Normally triggered when bees are nearby their attacker.)
Context: https://discordapp.com/channels/289587909051416579/925530366192779286/952619743490236426
image

Piglin get/setIsChargingCrossbow
image

Wolf get/setIsInterested
image

### CODE REVIEW

File: patches/api/0312-Missing-Entity-Behavior-API.patch [hunk 1]
```diff
@@ -69,6 +69,31 @@ index 0d88dce9978243a1f995c5fb448c5d71b01136eb..cad47139de57642fb3bb483e7a5acaa7
...
  }
+diff --git a/src/main/java/org/bukkit/entity/Bee.java b/src/main/java/org/bukkit/entity/Bee.java
+index adb20a9abba33c32d553f620fa82b27dff64ab5f..063ae8627f53c733b0f691a259e6062a223537af 100644
+--- a/src/main/java/org/bukkit/entity/Bee.java
++++ b/src/main/java/org/bukkit/entity/Bee.java
+@@ -93,4 +93,20 @@ public interface Bee extends Animals {
+      * @param ticks Ticks the bee cannot enter a hive for
+      */
+     void setCannotEnterHiveTicks(int ticks);
++
```
Comment (jpenilla) [inline patches/api/0312-Missing-Entity-Behavior-API.patch:80]: paper comments

**Review by Machine-Maker — APPROVED** [review]

File: patches/api/0312-Missing-Entity-Behavior-API.patch [hunk 2]
```diff
@@ -69,6 +69,39 @@ index 0d88dce9978243a1f995c5fb448c5d71b01136eb..cad47139de57642fb3bb483e7a5acaa7
...
++    // Paper start
++    /**
++     * Sets the override for if the bee is currently rolling.
++     *
++     * @param rolling is rolling, or unset for vanilla behavior
++     */
++    void setRollingOverride(@org.jetbrains.annotations.NotNull net.kyori.adventure.util.TriState rolling);
```
Comment (MiniDigger) [inline patches/api/0312-Missing-Entity-Behavior-API.patch:87]: would be cool to have a line here for what exactly its overriding, this is a bit confusing as is

File: patches/server/0676-Missing-Entity-Behavior-API.patch [hunk 3]
```diff
@@ -6,6 +6,24 @@ Subject: [PATCH] Missing Entity Behavior API
...
 
+diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
+index c49e7ea3e2efc4459f5ed1d4ebd83c9d23420611..79a9fa2a2257925685e45329365d2828ccd4c48c 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
++++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
+@@ -539,11 +539,13 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
+         this.setFlag(4, hasStung);
+     }
+ 
++    public net.kyori.adventure.util.TriState rollingOverride = net.kyori.adventure.util.TriState.NOT_SET; // Paper - Rolling override
```
Comment (MiniDigger) [inline patches/server/0676-Missing-Entity-Behavior-API.patch:17]: not tooooo sure if its a good idea to use the adventure tri state here but I guess adventure isn't going anywhere and its just internal

**Review by MiniDigger — APPROVED** [review]

File: patches/server/0676-Missing-Entity-Behavior-API.patch (hunk 3 above)
Comment (Machine-Maker) [reply patches/server/0676-Missing-Entity-Behavior-API.patch:17]: well its not just internal. that's set directly by the API. your other comment was right below where that happened.

Comment (MiniDigger) [reply patches/server/0676-Missing-Entity-Behavior-API.patch:17]: its early, I shouldn't review stuff this early...
I see now, idk, guess its fine

**Comment by molor** [comment]
Add Vex bound & life ticks setter please

**Comment by Owen1212055** [comment]
> Add Vex bound & life ticks setter please

Will do, thanks for the suggestion here. :)

**Comment by Owen1212055** [comment]
> Add Vex bound & life ticks setter please

Added new vex API + rebased

**Review by kennytv — APPROVED** [review]

**Comment by molor** [comment]
(why no method to change Vex's bound ):)

**Comment by Owen1212055** [comment]
> (why no method to change Vex's bound ):)

whoops, I forgor. I'll bundle that into another entity api PR.

**Pull request merged**